queries.sql

```diff
@@ -22,4 +22,88 @@ SELECT * FROM animals
 WHERE NOT name='Gabumon';
 
 SELECT * FROM animals
-WHERE weight_kg>=10.4 AND weight_kg<=17.3;
\ No newline at end of file
+WHERE weight_kg>=10.4 AND weight_kg<=17.3;
+
+/* In a Transaction update the animals table by setting the species column to unspecified */
+
+BEGIN;
+ALTER TABLE animals
+RENAME COLUMN species TO unspecified;
+
+/* Very that the change was made and roll back the change */
+SELECT * FROM animals;
+ROLLBACK;
+
+/* Inside a transaction Update the animals table by setting the species column to digimon for all animals that have a name ending in mon */
+
+BEGIN;
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+/* Update the animals table by setting the species column to pokemon for all animals that don't have species already set. */
+UPDATE animals
+SET species = 'pokemon'
+WHERE NOT name LIKE '%mon';
+
+/* Commit the transaction.*/
+COMMIT;
+
+/* Inside a transaction delete all records in the animals table, then roll back the transaction */
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+
+/* Inside a transaction Delete all animals born after Jan 1st, 2022 */
+BEGIN;
+
+DELETE FROM animals
+WHERE date_of_birth>'01-01-2022';
+
+/* Create a savepoint for the transaction. */
+SAVEPOINT SP1;
+
+/* Update all animals' weight to be their weight multiplied by -1. */
+UPDATE animals
+SET weight_kg = weight_kg*-1;
+
+/* Rollback to the savepoint */
+ROLLBACK TO SP1;
+
+U/* pdate all animals' weights that are negative to be their weight multiplied by -1. */
+UPDATE animals
+SET weight_kg = weight_kg*-1
+WHERE weight_kg<0;
+
+/* Commit transaction */
+COMMIT;
+
+/* How many animals are there? */
+SELECT COUNT(*) FROM animals;
+
+/* How many animals have never tried to escape? */
+SELECT COUNT(*) FROM animals
+vet_clinic-# WHERE escape_attempts=0;
+
+/* What is the average weight of animals? */
+SELECT AVG(weight_kg) FROM animals;
+
+/* Who escapes the most, neutered or not neutered animals? */
+SELECT MAX(escape_attempts) FROM animals;
+SELECT COUNT(name) FROM animals
+vet_clinic-# WHERE neutered = TRUE;
+SELECT COUNT(name) FROM animals
+vet_clinic-# WHERE neutered = FALSE;
+
+/* What is the minimum and maximum weight of each type of animal? */
+SELECT MIN(weight_kg) FROM animals
+vet_clinic-# GROUP BY species; 
+
+SELECT MAX(weight_kg) FROM animals
+vet_clinic-# GROUP BY species;
+
+/* What is the average number of escape attempts per animal type of those born between 1990 and 2000? */
+SELECT AVG(escape_attempts) FROM animals
+WHERE date_of_birth BETWEEN '01-01-1990' AND '01-01-2000'
```

Comment:
- [OPTIONAL] I think here, the date_of_birth >= '1990/01/01' AND date_of_birth <= '2000/12/31', please kindly correct that.